### PR TITLE
Cleanup a bit PBR master node - this fix issue with transparent shadow

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
Do some cleanup of the PBR master node, the shadow were not working with transparent.

+ add selection pass for PBR master node

---
### Testing status
**Katana Tests**: ca,'t launch katana test for now, too much load :(
**Manual Tests**: Try to compare a lit master node and a PBR master ndoe

---
### Overall Product Risks
**Technical Risk**: Low
**Halo Effect**:  Low

### CODE REVIEW

**Comment by spaztronaut** [comment]
Not seeing selection for mat using PBR master node: https://gyazo.com/887cd49185ca66fed21a1ab6a7c1588d

File: com.unity.render-pipelines.high-definition/Editor/Material/PBR/ShaderGraph/HDPBRPass.template [hunk 1]
```diff
@@ -24,15 +24,15 @@ Pass
...
+        #pragma multi_compile_instancing
+        #pragma instancing_options renderinglayer
+
+        #pragma multi_compile _ LOD_FADE_CROSSFADE
+
     //-------------------------------------------------------------------------------------
     // Variant Definitions (active field translations to HDRP defines)
     //-------------------------------------------------------------------------------------
         $AlphaTest:                      #define _ALPHATEST_ON 1
-        $Material.SubsurfaceScattering:  #define _MATERIAL_FEATURE_SUBSURFACE_SCATTERING 1
```
Comment (russoh) [inline com.unity.render-pipelines.high-definition/Editor/Material/PBR/ShaderGraph/HDPBRPass.template]: The corresponding code below this should go too:

#ifdef _MATERIAL_FEATURE_SUBSURFACE_SCATTERING
        surfaceData.materialFeatures |= MATERIALFEATUREFLAGS_LIT_SUBSURFACE_SCATTERING;
#endif
#ifdef _MATERIAL_FEATURE_TRANSMISSION
        surfaceData.materialFeatures |= MATERIALFEATUREFLAGS_LIT_TRANSMISSION;
#endif
#ifdef _MATERIAL_FEATURE_ANISOTROPY
        surfaceData.materialFeatures |= MATERIALFEATUREFLAGS_LIT_ANISOTROPY;
#endif
#ifdef _MATERIAL_FEATURE_CLEAR_COAT
        surfaceData.materialFeatures |= MATERIALFEATUREFLAGS_LIT_CLEAR_COAT;
#endif
#ifdef _MATERIAL_FEATURE_IRIDESCENCE
        surfaceData.materialFeatures |= MATERIALFEATUREFLAGS_LIT_IRIDESCENCE;
#endif

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/PBR/ShaderGraph/HDPBRPass.template]: Done

**Comment by sebastienlagarde** [comment]
>I did check the scene selection after reading Wyatt's comment - also doesn't work for me.
Dont' know why it doesn't work for you, it work for me.

graphnode

(same order than on your video @spaztronaut )

@RemyUnity can you do a test on your side here? thanks

**Comment by RemyUnity** [comment]
> > I did check the scene selection after reading Wyatt's comment - also doesn't work for me.
> 
> Dont' know why it doesn't work for you, it work for me.
> 
> graphnode
> 
> (same order than on your video @spaztronaut )
> 
> @RemyUnity can you do a test on your side here? thanks

I don't have the issue either.

**Comment by russoh** [comment]
Just resolve conflicts locally - the scene selection still isn't working for me. Using a simple cube with a default PBR shadergraph assigned.

**Comment by spaztronaut** [comment]
@sebastienlagarde @RemyUnity here's the project. you'll have to edit the paths in manifest.json.
[PR__misc-PBR-master-node-fix.zip](https://github.com/Unity-Technologies/ScriptableRenderPipeline/files/2485003/PR__misc-PBR-master-node-fix.zip)

**Comment by sebastienlagarde** [comment]
Thanks for the project, which version of the binaries are you using?

**Comment by spaztronaut** [comment]
> Thanks for the project, which version of the binaries are you using?

was using latest from 2018.3/staging

**Comment by spaztronaut** [comment]
I'll check out an earlier hash to see if this was something that broke in this PR (i don't want to keep it's merge held up)

**Comment by sebastienlagarde** [comment]
@zadi0648 @russoh So we have a problem. We have download your project and it run correctly for us... I have no idea what can go wrong, we have try with latest staging binaries and with 2018.3b6 and your project is fine. I have no clue about what's the problem. And as you are two to get the issue this is even more weird.

@Kink3d Would you be kind enough to try this branch with the linked project on your computer see if you have can reproduce the problem?

Thanks

**Comment by sebastienlagarde** [comment]
>Were the shadows not working in the last released package?
I don't see the relationship :), here this PR fix selection for PBR shader but it seems that it works for us in europe but doesn't for seattle people with the same project, so hard to figure out if there is an issue or not :). Need more people testing to validate.

**Comment by pbbastian** [comment]
Actually, since this only changes the HDRP sub-shader, the potential change log items should be in the HDRP change log.

**Comment by Kink3d** [comment]
@sebastienlagarde / @spaztronaut Whats the fate of this PR for 2019.1?

**Comment by sebastienlagarde** [comment]
This PR have been redo. Closong de have now  https://github.com/Unity-Technologies/ScriptableRenderPipeline/issues/2390

**Pull request closed without merging**